environment.cpp

```diff
@@ -35,7 +37,18 @@ bool operator== (const environment& lhs, const environment& rhs)
   bool ref_t_values = lhs.get_ref_target_values() == rhs.get_ref_target_values();
   bool current_t_value = are_equal_with_tolerance(lhs.get_current_target_value(), rhs.get_current_target_value());
 
-  return ref_t_values && current_t_value;
+  std::uniform_real_distribution<double> lhs_dist = lhs.get_cue_distribtion();
+  std::uniform_real_distribution<double> rhs_dist = rhs.get_cue_distribtion();
+
+  bool cue_distrib = are_same_distribution(lhs_dist, rhs_dist);
+
+  bool inputs = lhs.get_input() == rhs.get_input();
+
+  bool optimal = lhs.get_optimal() == rhs.get_optimal();
```

Comment:
Since `environment`s could have different rng from which they depend on, this might actually not be the same for two `environment`s that are more or less identical. Indeed having `optimal `and `inputs `as memeber variables is not exactly correct, as in my view they are notexactly part of the "internal state" of `environment`. we keep them there for convenience, but we might not want to check them. I don't know, your call

Reply:
Hmmm. I guess this depends in which context we're going to use the equality operator and what we are trying to check - are these two environments exact copies (in the same current state) or do they just have the same hard-coded parameters? I think we might need both at different times. So maybe let's keep it like this for now, and we can make a new function `have_same_parameters ` if the needs arise.